app/src/main/java/io/appium/settings/utils/GPSTracker.java

```diff
@@ -0,0 +1,109 @@
+/*
+  Copyright 2012-present Appium Committers
+  <p>
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+  <p>
+  http://www.apache.org/licenses/LICENSE-2.0
+  <p>
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+package io.appium.settings.utils;
+
+import android.app.Service;
+import android.content.Context;
+import android.content.Intent;
+import android.location.Location;
+import android.location.LocationListener;
+import android.location.LocationManager;
+import android.os.Bundle;
+import android.os.IBinder;
+import android.util.Log;
+
+public class GPSTracker extends Service implements LocationListener {
+    private static final String TAG = GPSTracker.class.getSimpleName();
+
+    private final Context mContext;
+
+    private static final long MIN_DISTANCE_CHANGE_FOR_UPDATES = 10; // 10 meters
+    private static final long MIN_TIME_BW_UPDATES = 1000 * 60; // 1 minute
+
+    public GPSTracker(Context context) {
+        this.mContext = context;
+    }
+
+    public Location getLocation() {
+        try {
+            final LocationManager locationManager =
+                    (LocationManager) mContext.getSystemService(LOCATION_SERVICE);
+            boolean isGPSEnabled = false;
+            try {
+                isGPSEnabled = locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER);
```

Comment:
I'd prefer to use [PASSIVE_PROVIDER](https://developer.android.com/reference/android/location/LocationManager.html#PASSIVE_PROVIDER)